### PR TITLE
Read from whitelist service

### PR DESCRIPTION
**Pull request by elizzhao**
Check to see if the user is on the whitelist created by the new tide service.

Code for tide service: 
https://github.com/DataBiosphere/tide/blob/master/src/server.js

### CODE REVIEW

File: src/libs/auth.js [hunk 1]
```diff
@@ -7,7 +7,6 @@ import { getConfig } from 'src/libs/config'
 import { reportError } from 'src/libs/error'
 import * as Utils from 'src/libs/utils'
 
-
```
Comment (zarsky-broad) [inline src/libs/auth.js]: Re-add this

File: src/libs/auth.js [hunk 2]
```diff
@@ -94,13 +93,12 @@ window.forceSignIn = async token => {
...
     clearNotification(sessionTimeoutProps.id)
-
+    const tideWhitelist =  await Ajax().Buckets.getObjectPreview('terra-tide-prod-data', 'whitelistEmails', undefined, true).then(res => res.json())
     Ajax().User.getStatus().then(response => {
       if (response.status === 404) {
         const isTrustedEmail = _.includes(state.user.email.match(/@.*/)[0],
           ['@broadinstitute.org', '@google.com', '@channing.harvard.edu', '@duke.corp-partner.google.com', '@stanford.corp-partner.google.com'])
-
-        if (getConfig().isProd && !isTrustedEmail && !ProdWhitelist.includes(md5(state.user.email))) {
+        if (getConfig().isProd && !isTrustedEmail && !ProdWhitelist.includes(md5(state.user.email)) && !tideWhitelist.includes(md5(state.user.email))) {
```
Comment (zarsky-broad) [inline src/libs/auth.js]: Re-add blank line above this.

File: src/libs/auth.js [hunk 3]
```diff
@@ -94,13 +93,12 @@ window.forceSignIn = async token => {
 authStore.subscribe(async (state, oldState) => {
   if (!oldState.isSignedIn && state.isSignedIn) {
     clearNotification(sessionTimeoutProps.id)
-
+    const tideWhitelist =  await Ajax().Buckets.getObjectPreview('terra-tide-prod-data', 'whitelistEmails', undefined, true).then(res => res.json())
```
Comment (zarsky-broad) [inline src/libs/auth.js]: Maybe only do this if the other checks for the user having access fail?

**Review by zarsky-broad — APPROVED** [review]

File: src/libs/ajax.js [hunk 4]
```diff
@@ -214,7 +214,7 @@ const User = signal => ({
   },
 
   uploadAttachment: async file => {
-    const res = await fetch(`https://broadinstitute.zendesk.com/api/v2/uploads?filename=${file.name}`, {
+    const res = await fetchOk(`https://broadinstitute.zendesk.com/api/v2/uploads?filename=${file.name}`, {
```
Comment (katevoss) [inline src/libs/ajax.js:217]: Why did you change this to `fetchOk`? The whitelist service doesn't seem related to zendesk attachments.
